### PR TITLE
Add procenv plugin for more detailed buildtime information

### PR DESCRIPTION
**Pull request by jcpunk**
This adds a plugin similar to hw_info.  Where hw_info produces detailed output on the hardware, this dumps a significant amount of information about the build time environment, any capabilities, and much more.

### CODE REVIEW

**Comment by jcpunk** [comment]
Thanks pep8!  I've fixed it.

File: mock/py/mockbuild/plugins/procenv.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @traceLog()
+    def _PreBuildHook(self):
+        getLog().info("enabled ProcEnv plugin")
+
+        self.buildroot.pkg_manager.install('procenv')
+
+        out_file = self.buildroot.resultdir + '/procenv.log'
+        with codecs.open(out_file, 'w', 'utf-8', 'replace') as out:
+
+            cmd = ["/usr/bin/procenv"]
```
Comment (xsuchy) [inline mock/py/mockbuild/plugins/procenv.py]: This add new dependence on package `procenv`. It needs to be added either as Requires or Suggests/Recommends

File: mock/etc/mock/site-defaults.cfg [hunk 2]
```diff
@@ -205,6 +205,9 @@
 # config_opts['plugin_conf']['hw_info_enable'] = True
 # config_opts['plugin_conf']['hw_info_opts'] = {}
 #
+# config_opts['plugin_conf']['procenv_enable'] = True
```
Comment (xsuchy) [inline mock/etc/mock/site-defaults.cfg]: 1) you should set the default in util.py as well (search for `hw_info_enable` there)
2) the information in procenv is too much detailed IMHO. I would propose to not enable it by default.

File: mock/py/mockbuild/plugins/procenv.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    ProcEnv(plugins, conf, buildroot)
+
+
+class ProcEnv(object):
+    # pylint: disable=too-few-public-methods
+    """Get the runtime process environment"""
+    @traceLog()
+    def __init__(self, plugins, conf, buildroot):
+        self.buildroot = buildroot
+        self.hw_info_opts = conf
```
Comment (xsuchy) [inline mock/py/mockbuild/plugins/procenv.py]: `hw_info_opts` - I guess this is left over from copy'n'paste from hw info plugin code :)

File: mock/py/mockbuild/plugins/procenv.py (hunk 1 above)
Comment (jcpunk) [reply mock/py/mockbuild/plugins/procenv.py]: Good call, I'll get that fixed.

File: mock/etc/mock/site-defaults.cfg (hunk 2 above)
Comment (jcpunk) [reply mock/etc/mock/site-defaults.cfg]: Works for me.  I'll confess I really like the extra information it provides... but I can turn it on where I'm using it easily enough.

File: mock/py/mockbuild/plugins/procenv.py (hunk 3 above)
Comment (jcpunk) [reply mock/py/mockbuild/plugins/procenv.py]: insert embarrassed emoji.

**Comment by jcpunk** [comment]
In theory, this push resolves the reported issues.

File: mock/etc/mock/site-defaults.cfg (hunk 2 above)
Comment (xsuchy) [inline mock/etc/mock/site-defaults.cfg]: This should be False. In other words, here it states what is the default.

**Comment by xsuchy** [comment]
Put before or after this line:
  https://github.com/jcpunk/mock/blob/procenv/mock/mock.spec#L80
line:
```
Suggests: procenv
```

**Comment by jcpunk** [comment]
The listed review items should be fixed now.

**Comment by xsuchy** [comment]
Thank you.
If you would document this plugin here https://github.com/rpm-software-management/mock/wiki#plugins you easy my life. Otherwise, I will do that during release.

**Pull request merged**